pkg/cmd/server/kubernetes/node/node_config.go

```diff
@@ -350,6 +352,17 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 			dnsConfig.Nameservers = options.DNSNameservers
 		}
 
+		// allow an upstream resolv.conf to override nameservers.
+		if len(options.DNSRecursiveResolvConf) > 0 {
+			c, err := miekgdns.ClientConfigFromFile(options.DNSRecursiveResolvConf)
+			if err != nil {
+				return nil, fmt.Errorf("could not start DNS, unable to read config file: %v", err)
+			}
+			for _, s := range c.Servers {
+				dnsConfig.Nameservers = append(dnsConfig.Nameservers, net.JoinHostPort(s, c.Port))
```

Comment:
I expected this to combine with `options.DNSNameservers` if specified, but override the defaults we get from `dns.NewServerDefaults()`